### PR TITLE
feat: Integrate cache decoder

### PR DESCRIPTION
**Pull request by zharinov**

### CODE REVIEW

File: bin/build-decoder.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -e
+
+pwd=$PWD
+deps=$pwd/deps
+
+cd "$deps/cache-decoder"
```
[the remark this answers is not in the document]
Comment (zharinov) [reply bin/build-decoder.sh:8]: I think decoder code can be categorized as dependency as well, while submodule and vendoring are just implementation details.

**Review by viceice — COMMENTED** [review]
can't we use a separate repo with GitHub pages, so this will get a known subdir of our docs?

**Comment by zharinov** [comment]
> can't we use a separate repo with GitHub pages, so this will get a known subdir of our docs?

No problem, I just wasn't sure whether it's possible for our docs to have pages from another repo. Transferred my repo to @rarkins

**Review by rarkins — APPROVED** [review]

**Comment by rarkins** [comment]
> can't we use a separate repo with GitHub pages, so this will get a known subdir of our docs?

Sorry, just saw this too late. We can refactor it if you prefer

**Comment by rarkins** [comment]
@zharinov should this URL be publicly available now?

**Pull request merged**